src/configurations.jl

```diff
@@ -89,22 +93,47 @@ function Configuration(orbs::Vector{Tuple{O,Int,Symbol}}) where {O<:AbstractOrbi
         push!(occupancy, occ)
         push!(states, state)
     end
-    Configuration(orbitals, occupancy, states)
+    Configuration(orbitals, occupancy, states, sorted)
 end
 
-Configuration(orbital::O, occupancy::Int, state::Symbol=:open) where {O<:AbstractOrbital} =
-    Configuration([orbital], [occupancy], [state])
+Configuration(orbital::O, occupancy::Int, state::Symbol=:open, sorted=true) where {O<:AbstractOrbital} =
+    Configuration([orbital], [occupancy], [state], sorted)
 
-Configuration{O}() where {O<:AbstractOrbital} =
-    Configuration(O[], Int[], Symbol[])
+Configuration{O}(sorted=true) where {O<:AbstractOrbital} =
+    Configuration(O[], Int[], Symbol[], sorted)
 
 const RelativisticConfiguration{N} = Configuration{RelativisticOrbital{N}}
 
-issimilar(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital} =
-    a.orbitals == b.orbitals && a.occupancy == b.occupancy
+"""
+    sort(cfg::Configuration)
+
+Returns a sorted copy of `cfg`.
+"""
+Base.sort(cfg::Configuration) =
+    Configuration(cfg.orbitals, cfg.occupancy, cfg.states)
+
+"""
+    sorted(cfg::Configuration)
+
+Returns `cfg` if `cfg.sorted` or a sorted copy otherwise.
+"""
+sorted(cfg::Configuration) =
+    cfg.sorted ? cfg : sort(cfg)
+
+function issimilar(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital}
+    ca = sorted(a)
+    cb = sorted(b)
+    ca.orbitals == cb.orbitals && ca.occupancy == cb.occupancy
+end
 
-Base.:(==)(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital} =
-    issimilar(a, b) && a.states == b.states
+function Base.:(==)(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital}
+    ca = sorted(a)
+    cb = sorted(b)
+    issimilar(ca, cb) && ca.states == cb.states
```

Comment:
Do we want two different configurations with everything except the ordering to be considered equal? Right now, that's the case.

Reply:
I would go with not considering them equal. E.g. if you couple (i.e. do the tree thing), differently ordered configurations would be coupled differently, so it would make sense not to consider the configurations to be the same.

I think `issimilar` is currently undocumented? Maybe document and export that, so it could be used when you want to compare configurations "qualitatively"?

Reply:
424111b